diff --git a/third_party/blink/renderer/modules/peerconnection/BUILD.gn b/third_party/blink/renderer/modules/peerconnection/BUILD.gn
index 054a691045e27ba624f4954a26a0f3e9a27af07e..0e1a7a93a551a053b342a35f6402e0c2210ea4dc 100644
--- a/third_party/blink/renderer/modules/peerconnection/BUILD.gn
+++ b/third_party/blink/renderer/modules/peerconnection/BUILD.gn
@@ -175,6 +175,7 @@ blink_modules_sources("peerconnection") {
 
   public_deps = [ "//third_party/webrtc_overrides:webrtc_component" ]
   deps = [
+    "//brave/components/brave_fingerprinting/mojom:mojom_blink",
     "//build:chromecast_buildflags",
     "//components/webrtc:net_address_utils",
     "//components/webrtc:thread_wrapper",
